lib/foreman_rh_cloud.rb

```diff
@@ -98,7 +98,18 @@ def self.transform_scheme(uri_string)
 
   # For testing purposes we can override the default hostname with an environment variable SATELLITE_RH_CLOUD_FOREMAN_HOST
   def self.foreman_host
-    @foreman_host ||= ::Host.unscoped.friendly.find(ENV['SATELLITE_RH_CLOUD_FOREMAN_HOST'] || ::SmartProxy.default_capsule.name)
+    @foreman_host ||= begin
+                        fullname = foreman_host_name
+                        ::Host.unscoped.friendly.find(fullname)
+                      rescue ActiveRecord::RecordNotFound
+                        # fullname didn't work. Let's try shortname
+                        shortname = /(?<shortname>[^\.]*)\.?.*/.match(fullname)[:shortname]
+                        ::Host.unscoped.friendly.find(shortname)
```

Comment:
in case the host's `shortname` wasn't found, should we log some error?

Reply:
`find` will throw an exception in case it's not found.

Reply:
Great, thanks!